chia/util/network.py

```diff
@@ -108,32 +109,25 @@ def class_for_type(type: NodeType) -> Any:
     raise ValueError("No class for type")
 
 
-def get_host_addr(host: Union[PeerInfo, str], prefer_ipv6: Optional[bool]) -> str:
-    # If there was no preference passed in (from config), set the system-wise
+def get_host_addr(host: str, prefer_ipv6: bool = False) -> IPAddress:
+    # If there was no preference passed in (from config), set the system-wide
     # default here.  Not a great place to locate a default value, and we should
     # probably do something to write it into the config, but.  For now...
-    if prefer_ipv6 is None:
-        prefer_ipv6 = False
-    # Use PeerInfo.is_valid() to see if it's already an address
-    if isinstance(host, PeerInfo):
-        hoststr = host.host
-        if host.is_valid(True):
-            return hoststr
-    else:
-        hoststr = host
-        if PeerInfo(hoststr, uint16(0)).is_valid(True):
-            return hoststr
+    try:
+        return ip_address(host)
+    except ValueError:
+        pass
     addrset: List[
         Tuple["socket.AddressFamily", "socket.SocketKind", int, str, Union[Tuple[str, int], Tuple[str, int, int, int]]]
-    ] = socket.getaddrinfo(hoststr, None)
+    ] = socket.getaddrinfo(host, None)
     # Addrset is never empty, an exception is thrown or data is returned.
     for t in addrset:
         if prefer_ipv6 and t[0] == socket.AF_INET6:
-            return t[4][0]
+            return ip_address(t[4][0])
         if not prefer_ipv6 and t[0] == socket.AF_INET:
-            return t[4][0]
+            return ip_address(t[4][0])
     # If neither matched preference, just return the first available
-    return addrset[0][4][0]
+    return ip_address(addrset[0][4][0])
```

Comment:
I realize you didn't really change this code, but it doesn't look correct. The list of IP addresses may contain multiple IPv4, multiple IPv6 and multiple of other address families. Just picking the first one here doesn't seem right. We should pick the first one that is either `AF_INET6` or `AF_INET`.

Maybe the loop should remember the first address of each type it finds, and then pick one of them that isn't `None` depending on the preference

Reply:
Agree and yeah it makes sense to adjust this while reworking this method, see [bf72dbd](https://github.com/Chia-Network/chia-blockchain/pull/14068/commits/bf72dbd25a500cc7cbbeb5742b21328193c759af).